### PR TITLE
Cache/CacheStorage secure contexts

### PR DESCRIPTION
**Pull request by hamishwillee**
This adds secure context header to `Cache` and `CacheStorage` docs and also fixes flaws.

### CODE REVIEW

File: files/en-us/web/api/cache/index.html [hunk 1]
```diff
@@ -21,15 +21,18 @@
...
 
-<div class="note">
-<p><strong>Note:</strong> The caching API doesn't honor HTTP caching headers.</p>
+<div class="notecard note">
+  <h4>Note</h4>
+  <p>The caching API doesn't honor HTTP caching headers.</p>
 </div>
 
 <p>{{AvailableInWorkers}}</p>
+<p>{{securecontext_header}}</p>
```
Comment (hamishwillee) [inline files/en-us/web/api/cache/index.html:35]: FYI Secure context is listed in BCD.

File: files/en-us/web/api/cachestorage/index.html [hunk 2]
```diff
@@ -29,9 +32,13 @@
 
 <p>You can access <code>CacheStorage</code> through the global {{domxref("WindowOrWorkerGlobalScope.caches", "caches")}} property.</p>
 
-<div class="note"><strong>Note</strong>: CacheStorage always rejects with a <code>SecurityError</code> on untrusted origins (i.e. those that aren't using HTTPS, although this definition will likely become more complex in the future.) When testing, you can get around this by checking the "Enable Service Workers over HTTP (when toolbox is open)" option in the Firefox Devtools options/gear menu.</div>
```
Comment (hamishwillee) [inline files/en-us/web/api/cachestorage/index.html:32]: @chrisdavidmills FYI added both available in workers and secure context header. 

I also deleted this section. I THINK that this essentially says this is a secure context :-). Further, the comment *, you can get around this by checking the "Enable Service Workers over HTTP (when toolbox is open)" option in the Firefox Devtools options/gear menu.* is probably something that should be in either securecontexts or serviceworker docs - not in each case it appears.

OK?

**Review by sideshowbarker — APPROVED** [review]

File: files/en-us/web/api/cachestorage/index.html (hunk 2 above)
Comment (sideshowbarker) [reply files/en-us/web/api/cachestorage/index.html:32]: (Added @chrisdavidmills as a reviewer, since this seems to be a needsinfo for him)

Comment (hamishwillee) [reply files/en-us/web/api/cachestorage/index.html:32]: Thanks @sideshowbarker - for some reason the system wouldn't let me add Chris myself.

Comment (sideshowbarker) [reply files/en-us/web/api/cachestorage/index.html:32]: > Thanks @sideshowbarker - for some reason the system wouldn't let me add Chris myself.

Yeah I think there may have been some Github glitch — I had trouble at first adding Chris too, but then suddenly… it just seemed to work

Comment (chrisdavidmills) [reply files/en-us/web/api/cachestorage/index.html:32]: Everything else looks good here, although I'd still keep this note. It does basically say (or heavily imply) that this requires a secure context, but it provides other useful information as well. What you you think?

**Review by chrisdavidmills — CHANGES_REQUESTED** [review]

File: files/en-us/web/api/cachestorage/index.html (hunk 2 above)
Comment (hamishwillee) [reply files/en-us/web/api/cachestorage/index.html:32]: @chrisdavidmills Thanks. I've added back the note; not completely convinced - so I also added it where I think it should be - in the service workers docs.

**Review by chrisdavidmills — APPROVED** [review]
Cool! Let's do this then.

**Pull request merged**